spacy/strings.pyx

```diff
@@ -110,28 +130,35 @@ cdef class StringStore:
         string_or_id (bytes, str or uint64): The value to encode.
         Returns (str / uint64): The value to be retrieved.
         """
-        if isinstance(string_or_id, str) and len(string_or_id) == 0:
-            return 0
-        elif string_or_id == 0:
-            return ""
-        elif string_or_id in SYMBOLS_BY_STR:
-            return SYMBOLS_BY_STR[string_or_id]
-        cdef hash_t key
+        cdef hash_t id
+        cdef Utf8Str* utf8str = NULL
+
         if isinstance(string_or_id, str):
-            key = hash_string(string_or_id)
-            return key
+            if len(string_or_id) == 0:
+                return 0
+
+            symbol = SYMBOLS_BY_STR.get(string_or_id, None)
+            if symbol is not None:
+                return symbol
+            else:
+                return hash_string(string_or_id)
         elif isinstance(string_or_id, bytes):
-            key = hash_utf8(string_or_id, len(string_or_id))
-            return key
-        elif string_or_id < len(SYMBOLS_BY_INT):
-            return SYMBOLS_BY_INT[string_or_id]
-        else:
-            key = string_or_id
-            utf8str = <Utf8Str*>self._map.get(key)
-            if utf8str is NULL:
-                raise KeyError(Errors.E018.format(hash_value=string_or_id))
+            return hash_utf8(string_or_id, len(string_or_id))
+        elif try_coerce_to_hash(string_or_id, &id):
+            if id == 0:
+                return ""
+            elif id < len(SYMBOLS_BY_INT):
+                return SYMBOLS_BY_INT[id]
             else:
-                return decode_Utf8Str(utf8str)
+                utf8str = <Utf8Str*>self._map.get(id)
+        else:
+            # TODO: Raise an error instead
```

Comment:
Not in this PR? Does this change the API?

Reply:
Maybe we should make a ticket for v4 to change this behavior?

Reply:
Yeah, that's the plan (just haven't manage to do it yet 😅).

Reply:
Do we know where the no-error behavior is relied on, if anywhere?